specutils/analysis/snr.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from ..spectra import SpectralRegion
+
+__all__ = ['snr']
+
+
+def snr(spectrum, region=None):
+    """
+    Calculate the mean S/N of the spectrum based on the flux and uncertainty
+    in the spectrum. This will be calculated over the regions, if they
+    are specified.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion` or list of `~specutils.utils.SpectralRegion`
```

Comment:
Something occurs to me here: as a convenience for users, perhaps we should also allow "anything that can be converted to a `SpectralRegion`?  That is, if you just pass it through the `SpectralRegion` initializer as ``SpectralRegion(*region), the user can do ``snr(spec, (12*u.pixel, 24*u.pixel)`` or similar, and have it work.  If you like I can just make a PR against your branch to look at, as I was testing with it locally anyway...

Reply:
In some out-of-band discussion, @brechmos-stsci and I discussed and agreed this could be a follow-on PR